tools/src/main/java/org/apache/kafka/tools/consumer/group/ShareGroupCommand.java

```diff
@@ -208,6 +222,67 @@ public void describeGroups() throws ExecutionException, InterruptedException {
             }
         }
 
+        Map<String, Throwable> deleteShareGroups() {
+            List<GroupListing> shareGroupIds = listDetailedShareGroups();
+            List<String> groupIds = opts.options.has(opts.allGroupsOpt)
+                ? shareGroupIds.stream().map(GroupListing::groupId).toList()
+                : opts.options.valuesOf(opts.groupOpt);
+
+            if (groupIds.isEmpty()) {
+                throw new IllegalArgumentException("--groups or --all-groups argument is mandatory");
```

Comment:
I would have said that the checking that there is either `--groups` or `--all-groups` should be done in the command line validation, not here.

As it stands, if the user specified `--all-groups` and there are no share groups, the error message is `--groups or --all-groups is mandatory`. Really, I would have said that this should be treated as successful deletion of no groups. That's what `kafka-consumer-groups.sh` does. It says "Deletion of requested share groups ('') was successful." Slight inelegant, but effective.